calyx/src/passes/well_formed.rs

```diff
@@ -78,6 +80,37 @@ where
     Ok(())
 }
 
+fn same_binding(binding_out: &SmallVec<[(ir::Id, u64); 5]>, binding_in:&SmallVec<[(ir::Id, u64); 5]>) -> bool {
```

Comment:
I would make this function return a CalyxResult instead so we can return better error messages for each case where we return false